java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/Partition.java

```diff
@@ -1,40 +1,50 @@
 package com.godaddy.asherah.appencryption;
 
-public class Partition {
+public abstract class Partition {
 
   private final String partitionId;
   private final String serviceId;
   private final String productId;
 
-  public Partition(final String partitionId, final String serviceId, final String productId) {
+  protected Partition(final String partitionId, final String serviceId, final String productId) {
     this.partitionId = partitionId;
     this.serviceId = serviceId;
     this.productId = productId;
   }
 
+  /**
+   * Gets the partitionId for a partition
+   * @return The partition id
+   */
   String getPartitionId() {
     return partitionId;
   }
 
+  /**
+   * Gets the serviceId for a partition
+   * @return  The service id
+   */
   String getServiceId() {
     return serviceId;
   }
 
+  /**
+   * Gets the productId for a partition
```

Comment:
for all the description javadoc, i think we can remove the word `partition` since the classname is `Partition`. So when you say `getServiceId`, it is implied that it is a service id for the partition.

Given that, we may be also able to remove the descriptions all together. Just return comments should be self explanatory.